### PR TITLE
Use builtin CA file if CA file is missing and we're using libressl

### PR DESCRIPTION
**Pull request by bassosimone**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -10,3 +10,4 @@ machine:
 test:
   override:
     - ./build/docker/valgrind-clang
+    - if test -f ./test-suite.log; then cat ./test-suite.log; fi
```
Comment (bassosimone) [inline circle.yml]: Strictly speaking this is not part of the pull request but was helpful to debug it.

**Comment by hellais** [comment]
I am going to review this following this:

* [ ] First pass at reviewing the code and add comments
* [ ] Local testing
* [ ] Merge

**Comment by hellais** [comment]
The code looks good to me. I added a comment on something that I think should be changed. After that is done I will do another pass and local testing.

File: src/libmeasurement_kit/net/ssl_context_impl.hpp [hunk 2]
```diff
@@ -34,26 +35,41 @@ static void initialize_ssl() {
...
+    if (path != "") {
+        if (!SSL_CTX_load_verify_locations(ctx, path.c_str(), nullptr)) {
+            debug("ssl: failed to load verify location");
+            SSL_CTX_free(ctx);
+            return SslCtxLoadVerifyLocationsError();
+        }
+    } else {
+#ifdef LIBRESSL_VERSION_NUMBER
+        std::vector<uint8_t> bundle = builtin_ca_bundle();
+        if (!SSL_CTX_load_verify_mem(ctx, bundle.data(), bundle.size())) {
```
Comment (bassosimone) [inline src/libmeasurement_kit/net/ssl_context_impl.hpp]: Good catch, thanks! I have implemented a conditional check such that we don't even attempt to use that function if we're under libressl < 2.1.4 (see ce5bd8e).

**Comment by bassosimone** [comment]
> After that is done I will do another pass and local testing.

For local testing, I guess it could help you to know the procedure I used:

```
make distclean
git clean -dffx
export pkg_make_flags="-jX" # where X is the parallelism you want
./build/dependency libressl    # 6 minutes with 8 processors
./build/dependency libevent   # 2 minutes with 8 processors
./build/dependency geoip      # 1 minute 
./autogen.sh
./configure --with-openssl=builtin --with-libevent=builtin --with-geoip=builtin --with-ca-bundle=""
make V=0 -jX  # again X is parallelism
./measurement_kit http_request -vv https://www.google.com/
# output should mention that is using libressl bundle ca and it should work as opposed to erroring out
make V=0 -jX ./test/net/ssl_context
./test/net/ssl_context -s
# it should work and output should mention using `SSL_CTX_load_verify_mem`
```

Thanks!

**Comment by hellais** [comment]
Ok I did some local smoke testing and the new updated code looks good.

You can proceed with merge once the CI tests pass!

**Comment by bassosimone** [comment]
> You can proceed with merge once the CI tests pass!

Epic, thank you!
